### ISSUE TITLE
Add point that reset variables must be initialised?

### ISSUE THREAD

**Issue by kerimoyle** [cellml/cellml-specification#329]
What do people feel about adding a point to either:
[- 2.9 The reset element](https://cellml-specification-dev.readthedocs.io/en/latest/reference/formal_and_informative/specB09.html), or
[- 3.11 Interpretation of resets](https://cellml-specification-dev.readthedocs.io/en/latest/reference/formal_and_informative/specC11_interpretation_of_variable_resets.html)
that says that the reset variable must be initialised?  I think it would make a huge amount of the operation of the resets clearer, it's easy to validate, and the resets won't actually work without it ...?

**Comment by MichaelClerx** [comment]
How do you mean "it must be initialised" ?

**Comment by kerimoyle** [comment]
It must have an `initial_value` attribute somewhere in the set?  Or have I got that wrong? ie: you can't define its value through the maths `A = 1` if you want to reset `A` anywhere ... ?

**Comment by MichaelClerx** [comment]
Yeah I think that that's the only way the reset rules make sense. We can't have both "x is always 3" and "set x to 7". Probably wherever we explain the difference between eternal truths and initial values would be the place to bring this up?

**Comment by kerimoyle** [comment]
Another confusing thing: It's not that initial values are only true at t=0, it's that they're true until changed (whether by integration, maths, or resets makes no difference).  The "initial" throws me a little, as I associate it only with integration.  Maybe we can use the attribute `initial_value` still, but refer to them in the text by something more generic, as we do with `map_variables` etc ...

**Comment by MichaelClerx** [comment]
That's cool!
We're only just working out what these things mean. It started with people saying "what constructs do we need to encode all the models we care about", I think, but now we're approaching it from the other end and saying "now that we've said X, what does that imply"? I like it

**Comment by MichaelClerx** [comment]
Initially, x=0, until changed by a reset or ODE ?

**Comment by kerimoyle** [comment]
... or a change in another variable in an equation? Anything that has a value that can change for any reason needs to be initialised directly through either `initial_value` or indirectly by other variables that it depends on being initialised?  I've started trying to figure this out ... see: https://libcellml-tutorials.readthedocs.io/en/pr344_documentation/tutorials/asides/aside_constant_vs_variable.html (work in progress!!)

**Comment by MichaelClerx** [comment]
> ... or a change in another variable in an equation?

how so?

**Comment by kerimoyle** [comment]
A = B + C
B is changed by a reset or ODE? I guess A doesn't need to be initialised then ...

> Initially, x=0, until changed by a reset or ODE ?

Can we not just say: x=0 until changed by something? Ditch the initial part?

**Comment by MichaelClerx** [comment]
Is "initial" not a shorthand for "when the vector t is some unspecified value t0" ?

**Comment by kerimoyle** [comment]
It implies that it's a state variable (to me anyway), rather than just a regular variable that you need to give a value to.  But I guess this could be cleared up with some clarity in whatever section in chapter 3 of the normative, and some good examples in the informative.
